mobile/lib/shared/services/sync.service.dart

```diff
@@ -0,0 +1,592 @@
+import 'dart:async';
+
+import 'package:collection/collection.dart';
+import 'package:flutter/foundation.dart';
+import 'package:hooks_riverpod/hooks_riverpod.dart';
+import 'package:immich_mobile/shared/models/album.dart';
+import 'package:immich_mobile/shared/models/asset.dart';
+import 'package:immich_mobile/shared/models/exif_info.dart';
+import 'package:immich_mobile/shared/models/store.dart';
+import 'package:immich_mobile/shared/models/user.dart';
+import 'package:immich_mobile/shared/providers/db.provider.dart';
+import 'package:immich_mobile/utils/diff.dart';
+import 'package:immich_mobile/utils/tuple.dart';
+import 'package:isar/isar.dart';
+import 'package:openapi/api.dart';
+import 'package:photo_manager/photo_manager.dart';
+
+final syncServiceProvider =
+    Provider((ref) => SyncService(ref.watch(dbProvider)));
+
+class SyncService {
+  final Isar _db;
+  Completer<bool> _remoteAlbumCompleter = Completer()..complete(false);
+  Completer<bool> _remoteAssetCompleter = Completer()..complete(false);
+  Completer<bool> _deviceCompleter = Completer()..complete(false);
+
+  SyncService(this._db);
+
+  // public methods:
+
+  /// Syncs users from the server to the local database
+  /// Returns `true`if there were any changes
+  Future<bool> syncUsersFromServer(List<User> users) async {
+    users.sortBy((u) => u.id);
```

Comment:
Why do we need to sort user by id here? I believe this is the UUID from the server's database, correct?

Reply:
Correct, this is the UUID.
We need to sort the list in order to compare it with the list of all users currently in the Isar DB. `diffSortedListsSync` excepts two lists sorted by the same key